app/src/main/java/com/kylecombes/micrepeater/WelcomeActivity.java

```diff
@@ -34,37 +23,48 @@ protected void onCreate(Bundle savedInstanceState) {
         pagerAdapter = new ScreenSlidePagerAdapter(this);
         viewPager.setAdapter(pagerAdapter);
 
-        final Button backbutton = (Button) findViewById(R.id.back_button);
-        final Button nextbutton = (Button) findViewById(R.id.next_button);
-        backbutton.setOnClickListener(new View.OnClickListener() {
+        final Button backButton = findViewById(R.id.back_button);
+        final Button nextButton = findViewById(R.id.next_button);
+
+        viewPager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
+            public void onPageScrollStateChanged(int state) {
+            }
+
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+            }
+
+            public void onPageSelected(int position) {
+                if (viewPager.getCurrentItem() == 0) {
+                    // If the user returns to the first page, then the back button turns invisible
+                    backButton.setVisibility(View.INVISIBLE);
+                } else if (viewPager.getCurrentItem() == 4) {
+                    // If the user goes to the last page, but then returns to the pages before, the "next" button should read next instead of finish
+                    nextButton.setText(R.string.next);
+                } else if (viewPager.getCurrentItem() == 5) {
```

Comment:
What if we used `layout_files.length-1` instead of hardcoding `5` here? (Also, since `layout_files` is a constant, maybe it should be `LAYOUT_FILES`?)

Reply:
Fixed